changeset:   5428:d8bf0c2cfab0
branch:      HEAD
tag:         tip
user:        dylan@datura.dylex.net
date:        Thu Jul 15 15:48:08 2010 -0400
summary:     Add new status sort method

diff -r 6b13e8319c1f -r d8bf0c2cfab0 init.h
--- a/init.h	Fri Jul 09 11:07:18 2010 -0700
+++ b/init.h	Thu Jul 15 15:48:08 2010 -0400
@@ -3368,6 +3368,7 @@
   { "to",		SORT_TO },
   { "score",		SORT_SCORE },
   { "spam",		SORT_SPAM },
+  { "status",		SORT_STATUS },
   { NULL,               0 }
 };
 
@@ -3387,6 +3388,7 @@
   { "to",		SORT_TO },
   { "score",		SORT_SCORE },
   { "spam",		SORT_SPAM },
+  { "status",		SORT_STATUS },
   { NULL,               0 }
 };
 
diff -r 6b13e8319c1f -r d8bf0c2cfab0 sort.c
--- a/sort.c	Fri Jul 09 11:07:18 2010 -0700
+++ b/sort.c	Thu Jul 15 15:48:08 2010 -0400
@@ -210,6 +210,20 @@
   return (SORTCODE(result));
 }
 
+static int compare_status (const void *a, const void *b)
+{
+  HEADER **pa = (HEADER **) a;
+  HEADER **pb = (HEADER **) b;
+  int result = 0;
+  if (!result) result = (*pb)->deleted - (*pa)->deleted;
+  if (!result) result = (*pa)->flagged - (*pb)->flagged;
+  if (!result) result = (*pb)->replied - (*pa)->replied;
+  if (!result) result = (*pb)->old - (*pa)->old;
+  if (!result) result = (*pb)->read - (*pa)->read;
+  AUXSORT(result,a,b);
+  return (SORTCODE(result));
+}
+
 sort_t *mutt_get_sort_func (int method)
 {
   switch (method & SORT_MASK)
@@ -232,6 +246,8 @@
       return (compare_score);
     case SORT_SPAM:
       return (compare_spam);
+    case SORT_STATUS:
+      return (compare_status);
     default:
       return (NULL);
   }
diff -r 6b13e8319c1f -r d8bf0c2cfab0 sort.h
--- a/sort.h	Fri Jul 09 11:07:18 2010 -0700
+++ b/sort.h	Thu Jul 15 15:48:08 2010 -0400
@@ -31,6 +31,7 @@
 #define SORT_KEYID	12
 #define SORT_TRUST	13
 #define SORT_SPAM	14
+#define SORT_STATUS	15
 /* dgc: Sort & SortAux are shorts, so I'm bumping these bitflags up from
  * bits 4 & 5 to bits 8 & 9 to make room for more sort keys in the future. */
 #define SORT_MASK	0xff

